debian/rules

```diff
@@ -1,114 +1,50 @@
 #!/usr/bin/make -f
-# -*- makefile -*-
-# Sample debian/rules that uses debhelper.
-# GNU copyright 1997 to 1999 by Joey Hess.
 
 # Uncomment this to turn on verbose mode.
 #export DH_VERBOSE=1
 
-CFLAGS = -Wall -Wno-format -g
-
 ifneq (,$(findstring noopt,$(DEB_BUILD_OPTIONS)))
-	CFLAGS += -O0
+	CFLAGS_MAINT += -O0
 else
-	CFLAGS += -O2
+	CFLAGS_MAINT += -O2
 endif
 ifeq (,$(findstring nostrip,$(DEB_BUILD_OPTIONS)))
 	INSTALL_PROGRAM += -s
 endif
 ifneq (,$(findstring thumb,$(DEB_BUILD_OPTIONS)))
-        CFLAGS += -mthumb
+	CFLAGS_MAINT += -mthumb
 endif
 ifeq (,$(findstring nolauncher,$(DEB_BUILD_OPTIONS)))
 	CONFIGURE_OPTIONS += --enable-maemo-launcher
 	USE_MAEMO_LAUNCHER = true
 endif
 
-CFLAGS += -DTAP_AND_HOLD
-
-CXXFLAGS = $(CFLAGS)
-
-config.status:
-	if [ ! -x configure ]; then ./autogen.sh; fi
-	dh_testdir
-	# Add here commands to configure the package.
-	CFLAGS="$(CFLAGS)" CXXFLAGS="$(CXXFLAGS)" ./configure --prefix=/usr --sysconfdir=/etc --mandir=\$${prefix}/share/man --infodir=\$${prefix}/share/info --disable-static $(CONFIGURE_OPTIONS)
-
-
-build: build-stamp
+CONFIGURE_OPTIONS += --disable-static
 
-build-stamp:  config.status
-	dh_testdir
+CPPFLAGS_MAINT += -DTAP_AND_HOLD
+CXXFLAGS_MAINT += -Wall -Wno-format -Wno-format-security -g
+LDFLAGS_MAINT += -Wl,--as-needed
 
-	# Add here commands to compile the package.
-	$(MAKE)
+export DEB_CPPFLAGS_MAINT_APPEND = $(CPPFLAGS_MAINT)
+export DEB_CXXFLAGS_MAINT_APPEND = $(CXXFLAGS_MAINT)
+export DEB_CFLAGS_MAINT_APPEND   = $(CFLAGS_MAINT)
+export DEB_LDFLAGS_MAINT_APPEND  = $(LDFLAGS_MAINT)
 
-	touch build-stamp
+%:
+	dh $@
```

Comment:
Let's move this part to the top of file, like it's done in dh manpage.
I think all commented line in the head of file may be removed.

Reply:
I would rather keep it as it is, having the if logic and exports before the actual target. It makes more sense whe looking at it procedurally.